doc/user/flycheck-versus-flymake.rst

```diff
@@ -44,71 +37,68 @@ Please do **not** use this table alone to make your personal judgment.  Read the
+|selection <Manual          |                       |                       |
+|Selection_>`_              |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Multiple syntax checkers  |yes                    |yes                    |
+|per buffer`_               |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Supported languages`_     |>50                    |2 [#]_                 |
+|                           |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|Checking remote files      |said to work, but not  |partly?                |
+|via Tramp                  |officially supported   |                       |
+|                           |[#]_                   |                       |
++---------------------------+-----------------------+-----------------------+
+|`Definition of new         |single declarative     |function definition    |
+|syntax checkers`_          |function/macro         |[#]_                   |
++---------------------------+-----------------------+-----------------------+
+|`Error levels`_            |errors, warnings,      |errors, warnings,      |
+|                           |infos, custom levels   |notes, custom levels   |
+|                           |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|`Error identifiers`_       |yes                    |no                     |
++---------------------------+-----------------------+-----------------------+
+|`Error parsing`_           |can use arbitrary      |can use arbitrary      |
+|                           |functions; helpers for |parsing functions; no  |
+|                           |regexp, JSON and XML   |built-in helpers       |
++---------------------------+-----------------------+-----------------------+
+|Error highlighting in      |yes                    |yes                    |
+|buffers                    |                       |                       |
++---------------------------+-----------------------+-----------------------+
+|Fringe icons for errors    |yes                    |yes (Emacs 24.1+)      |
++---------------------------+-----------------------+-----------------------+
+|`Error message display`_   |tooltip, echo area,    |tooltip only [#]_      |
+|                           |fully customizable     |                       |
++---------------------------+-----------------------+-----------------------+
+|List of all errors         |yes                    |yes                    |
++---------------------------+-----------------------+-----------------------+
+|`Resource consumption`_    |low                    |high                   |
```

Comment:
What do you base this on? Any benchmarks? Because implementation-wise, flymake is just as lazy as flycheck is regarding not visible buffers, for example.

Reply:
I'm bugged by this too, especially if this is an unfair characterization of the new Flymake.  (I wasn't the original author of the comparison, BTW).  I haven't had time to profile or check if [these sections](http://www.flycheck.org/en/1403-update-flymake-comparison/user/flycheck-versus-flymake.html#resource-consumption) are still accurate for Flymake.

The only thing I did notice using the Emacs Lisp backend was that inserting many newlines (by holding `C-o`) triggered many syntax checks, which ultimately froze Emacs for multiple seconds. So, basically, it looked like there was no deferring of checks in the case of newlines.

Reply:
Oh crud, I lost my comment. I don't know if you read it. I explained that the check-on-newline behaviour is configurable (it's on by default, maybe it shouldn't be). The variable is `flymake-start-check-on-newline` or sth. If you turn it off it's just as lazy as flycheck.